drivers/gpu/drm/vc4/vc4_hdmi.c

```diff
@@ -1261,6 +1261,35 @@ vc4_hdmi_encoder_clock_valid(const struct vc4_hdmi *vc4_hdmi,
 	return MODE_OK;
 }
 
+static unsigned long
+vc4_hdmi_encoder_compute_mode_clock(const struct drm_display_mode *mode,
+				    unsigned int bpc)
+{
+	unsigned long clock = mode->crtc_clock * 1000;
+
+	if (mode->flags & DRM_MODE_FLAG_DBLCLK)
+		clock = clock * 2;
+
+	return clock * bpc / 8;
```

Comment:
This overflows resulting in scrambling not getting enabled. (e.g. `clock=594000000` `bpc=8`)
Need something like `return (u64)clock * bpc / 8`